File: class2relational/src/main/java/org/atlanmod/model/scratch/relationalModel/RelationalModel.java
package org.atlanmod.model.scratch.relationalModel;

import org.atlanmod.tl.model.Model;
import scala.collection.immutable.List;

public class RelationalModel implements Model<RelationalElement, RelationalLink> {

    private RelationalMetamodel metamodel;
    private List<RelationalElement> elements;
    private List<RelationalLink> links;

    public RelationalModel(List<RelationalTable> tables) {
        // TODO
        this.metamodel = new RelationalMetamodel();
    }


    @Override
    public List<RelationalElement> allModelElements() {
        //TODO
        return null;
    }

    @Override
    public List<RelationalLink> allModelLinks() {
        //TODO
        return null;
    }
}

File: relational2class/src/main/scala/org/atlanmod/model/relationalmodel/RelationalModel.scala
package org.atlanmod.model.relationalmodel

import org.atlanmod.model.DynamicModel

class RelationalModel(elements: List[RelationalElement] = List(), links: List[RelationalLink] = List())
  extends DynamicModel(elements, links){

    override def allModelElements: List[RelationalElement] = elements
    override def allModelLinks: List[RelationalLink] = links

}